src/Tools/DynamoInstallDetective/ProductLookUp.cs

```diff
@@ -158,8 +161,16 @@ static string GetDisplayName(RegistryKey key)
         /// <param name="fileLookup">file name pattern to lookup</param>
         public InstalledProductLookUp(string lookUpName, string fileLookup)
         {
-            this.ProductLookUpName = lookUpName;
-            this.fileLocator = (path) => Directory.GetFiles(path, fileLookup).FirstOrDefault();
+            ProductLookUpName = lookUpName;
+            fileLocator = (path) =>
+            {
+                if (string.IsNullOrEmpty(corePath))
+                {
+                    corePath = Directory.GetFiles(path, fileLookup, SearchOption.AllDirectories)
```

Comment:
why don't you use `enumerateFiles` which will return an Ienumerable, then when you find the first matching file you can stop the lookup instead of returning all matching files which will require the search to march all the nested subdirectories even if you already found the file.
https://docs.microsoft.com/en-us/dotnet/api/system.io.directory.enumeratefiles?view=netframework-4.8